### DISCUSSION TITLE
Simple way to programmatically get stats from running instance?

### DISCUSSION THREAD

**Discussion by jeff-luszcz** [jvde-github/AIS-catcher#112] [Q&A] [answered]
I'd like to query my running instance and get information list that which is displayed in the statistic web interface.
Primarily I'd like to get the current Max distance seen and max messages per hour.

The goal is to make an alert for times where tropospheric propagation is ongoing (where my AIS captures move from an average of 9 miles to 20 miles or more (has 113 mile captures 2 weeks ago during a period of heavy tropospheric propagation!)

I'd rather not capture ever packet and store info / do math, if possible.
I'm running A/is-catcher as a service using ABCDE's great scripts.

Thoughts on the best way to approach this?

Reading the stats.bin file?
Other API?

**Comment by jvde-github** [comment, marked as answer]
Hi Jeff,

Do you have the webserver functionality running? That produces JSON files that are shown by the web interface and can be easily downloaded. 

For example the STAT page has the content from stat.json, which can be retreived in Linux like:
```
wget raspberrypi:8100/stat.json
cat stat.json | gunzip
```
and the ships (with their distance):
```
wget raspberrypi:8100/ships.json
```
Here raspberrypi:8100 is the address of my web interface. 

You can also in your browser go to: raspberrypi:8100/ships.json

Is that what you are after?

**Reply by jeff-luszcz** [reply]
Very helpful, great functionality, this should get be exactly what I need. thanks!

**Comment by tkreyche** [comment]
Hi Jeff,

 

Great idea but you need a more sophisticated algorithm than an average. For example, in my area there are many vessels within 5 miles. If a few distant ones are received under unusual atmospheric conditions, it will barely cause a blip on the average and would probably be in a normal noise level.

 

One option would be to sort vessels in bins by distance, maybe bins of 5-10, 10-20 miles etc. The bins don’t have to be consistent intervals. You also might want a bin to capture bad gps records, say > 500 miles.

Then you count the number of records in each bin, and create an alert based on some condition. It would take some testing to implement a universal rule, since every station location is going to have very different distributions. 

 

I’m sure there are better algorithms but I’m caffeine deficient right now. 

 

…Tom Kreyche

 

From: Jeff Luszcz ***@***.***> 
Sent: Wednesday, May 24, 2023 8:29 AM
To: jvde-github/AIS-catcher ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [jvde-github/AIS-catcher] Simple way to programmatically get stats from running instance? (Discussion #112)

 

I'd like to query my running instance and get information list that which is displayed in the statistic web interface.
Primarily I'd like to get the current Max distance seen and max messages per hour.

The goal is to make an alert for times where tropospheric propagation is ongoing (where my AIS captures move from an average of 9 miles to 20 miles or more (has 113 mile captures 2 weeks ago during a period of heavy tropospheric propagation!)

I'd rather not capture ever packet and store info / do math, if possible.
I'm running A/is-catcher as a service using ABCDE's great scripts.

Thoughts on the best way to approach this?

Reading the stats.bin file?
Other API?

**Comment by jvde-github** [comment]
For completeness :-) the input for the graphs tab is in:
```
http://raspberrypi:8100/history_full.json
```

**Comment by jeff-luszcz** [comment]
Thank so much.
I used this information to make a utility to send Mastodon alerts for instances of possible Tropospheric Ducting which might be of interest to Amateur Radio operators on VHF/UHF

Basically if I see AIS reports more than 10 (or 20 or 30) miles from my station, this indicates possible Tropospheric Ducting, a time where your radio signal might go much further than usual. I'll check every 15 minutes and send an alert if my received signals are further than a specified distance.

Code repository here:
https://github.com/jeff-luszcz/AISTropoAlert

Mastodon bot account here:
https://mastodon.social/@AISTropoAlertBoston

Thanks again for the help!
